logstash-core/spec/logstash/util/wrapped_synchronous_queue_spec.rb

```diff
@@ -3,26 +3,65 @@
 require "logstash/util/wrapped_synchronous_queue"
 
 describe LogStash::Util::WrappedSynchronousQueue do
- context "#offer" do
-   context "queue is blocked" do
-     it "fails and give feedback" do
-       expect(subject.offer("Bonjour", 2)).to be_falsey
-     end
-   end
-
-   context "queue is not blocked" do
-     before do
-       @consumer = Thread.new { loop { subject.take } }
-       sleep(0.1)
-     end
-
-     after do
-       @consumer.kill
-     end
-     
-     it "inserts successfully" do
-       expect(subject.offer("Bonjour", 20)).to be_truthy
-     end
-   end
- end
+  context "#offer" do
+    context "queue is blocked" do
+      it "fails and give feedback" do
+        expect(subject.offer("Bonjour", 2)).to be_falsey
+      end
+    end
+
+    context "queue is not blocked" do
+      before do
+        @consumer = Thread.new { loop { subject.take } }
+        sleep(0.1)
+      end
+
+      after do
+        @consumer.kill
+      end
+
+      it "inserts successfully" do
+        expect(subject.offer("Bonjour", 20)).to be_truthy
+      end
+    end
+  end
+
+  describe "queue clients" do
+    context "when requesting a write client" do
+      it "returns a client" do
+        expect(subject.write_client).to be_a(LogStash::Util::WrappedSynchronousQueue::WriteClient)
+      end
+    end
+
+    context "when requesting a read client" do
+      it "returns a client" do
+        expect(subject.read_client).to be_a(LogStash::Util::WrappedSynchronousQueue::ReadClient)
+      end
+    end
+
+    class Que < Array
+      def take() shift(); end
+      def poll(*) shift(); end
+    end
+
+    describe "WriteClient | ReadClient" do
+      context "when writing to the queue" do
+        let(:queue) { Que.new }
```

Comment:
It might be easier on readers to make it more verbose. `DummyQueue` would be more descriptive be more aligned with other tests the logstash codebase.